patroni/ha.py

```diff
@@ -1571,6 +1575,17 @@ def demote(self, mode: str) -> Optional[bool]:
 
         status = {'released': False}
 
+        demote_cluster_with_archive = False
+        archive_cmd = self.state_handler.get_archive_command()
+        if mode == 'demote-cluster':
+            # We need to send the shutdown checkpoint WAL file to archive to eliminate the need of rewind
+            # from a promoted instance that was previously replicating from archive
+            # When doing this, we disable stop timeout and do not run on_shutdown callback
+            if archive_cmd is None:
+                logger.info('Not archiving latest checkpoint WAL file. Archiving is not configured.')
```

Comment:
I think this message shows up in logs to early.
Better to do it after shutdown.